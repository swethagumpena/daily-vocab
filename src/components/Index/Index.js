import React from 'react';
import { useHistory } from 'react-router-dom';
import styles from './Index.module.scss';

const Index = () => {
  const history = useHistory();
  return (
    <div data-testid="Index" className={styles.page}>
      <div className={styles.container}>
        <h3>daily vocab!</h3>
        <p>learn a new  word everyday!</p>
        <button type="button" onClick={() => history.push('/home')}>Get Started</button>
      </div>
    </div>
  );
};

export default Index;
