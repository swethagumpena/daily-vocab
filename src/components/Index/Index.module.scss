@import "../../styles/index.scss";

.page {
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-image: linear-gradient(#ADD2FF, white);
}

.container{
    display: flex;
    flex-direction: column;
    align-items: center;
}

.container h3 {
    font-family: "Inconsolata", monospace;
    font-size: 4.5rem;
    letter-spacing: 0.25vw;
    margin: 1rem;
    color: $primary-text-color;
}
 
.container p {
    font-family: "Inconsolata", monospace;
    font-size: 2rem;
    letter-spacing: 0.025vw;
}

.container button {
    font-family: "Inconsolata", monospace;
    padding: 1rem;
    border-radius: 2rem;
    border: white;
    background-color: #ADD2FF;
    color: $primary-text-color;
    font-weight: bold;
}

@include media-mixin($mobile-breakpoint){
    .container h3{
        font-size: 3rem;
    }
    .container p{
        font-size: 1.5rem;
    }
    .container button{
        padding: 1rem;
    }
}
