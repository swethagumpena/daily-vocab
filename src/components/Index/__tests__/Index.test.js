import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Index from '../Index';

describe(Index.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<Index />);
    expect(container).toMatchSnapshot();
  });

  test('should display daily vocab!', () => {
    render(<Index />);
    expect(screen.getByText('daily vocab!'));
  });

  test('should display learn a new word everyday!', () => {
    render(<Index />);
    expect(screen.getByText('learn a new word everyday!'));
  });

  test('should display a Get  started button', () => {
    render(<BrowserRouter><Index /></BrowserRouter>);
    const gettingStartedElement = screen.getByText('Get Started');
    expect(gettingStartedElement.tagName).toBe('BUTTON');
    fireEvent.click(gettingStartedElement);
    expect(document.location.href).toBe('http://localhost/home');
  });
});
