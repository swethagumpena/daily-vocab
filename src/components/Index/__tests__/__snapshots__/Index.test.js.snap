// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
<div>
  <div
    class="page"
    data-testid="Index"
  >
    <div
      class="container"
    >
      <h3>
        daily vocab!
      </h3>
      <p>
        learn a new  word everyday!
      </p>
      <button
        type="button"
      >
        Get Started
      </button>
    </div>
  </div>
</div>
`;
